project/JarExtractor.scala

```diff
@@ -0,0 +1,178 @@
+import sbt.util.Logger
+
+import java.io.IOException
+import java.nio.file.{FileSystems, Files, Path}
+import java.util.jar.{JarEntry, JarFile, JarOutputStream}
+import scala.util.Using
+
+/** A jar entry can either be:
+  * - Copied to the output (thin) jar file.
+  * - Copied to the "extracted files" directory. These are usually native libs.
+  * - Skipped.
+  *
+  * If a jar entry does not match any of the globs, it is skipped.
+  * @param mapping Mapping of globs to commands for extracting files from a JAR.
+  */
+case class JarExtractor(
+  mapping: (String, JarExtractor.Command)*
+)
+
+object JarExtractor {
+
+  /** All supported native library architectures.
+    */
+  sealed trait NativeLibArch {
+    // Path of the library inside the extracted files directory
+    // Inspired by `org.enso.pkg.NativeLibraryFinder`
+    val path: String
+  }
+
+  case object LinuxX86_64 extends NativeLibArch {
+    override val path: String = "amd64/linux"
+  }
+  case object WindowsX86_64 extends NativeLibArch {
+    override val path: String = "amd64/windows"
+  }
+  case object MacOSX86_64 extends NativeLibArch {
+    override val path: String = "amd64/macos"
+  }
+  case object MacOSArm64 extends NativeLibArch {
+    override val path: String = "aarch64/macos"
+  }
+
+  // What to do with the matching jar entries.
+  sealed trait Command
+
+  case object CopyToOutputJar extends Command
+
+  /** @param arch If specified, will be copied only iff the architecture is
+    *        the same as the current platform. If not specified, will be copied
+    *        to the root of the extracted files directory.
```

Comment:
If specified, where the file is copied to? It is not clear from the description. Some examples would make it easier to understand.

Reply:
Example introduced in 91e13c88c76d1b3330c10025adbee4e9119e4b81 - inside https://github.com/enso-org/enso/blob/91e13c88c76d1b3330c10025adbee4e9119e4b81/project/JarExtractor.scala#L52-L53